tests/git.zunit

```diff
@@ -1263,6 +1263,675 @@
+        assert ${lines[3]} same_as "$(tput setaf 3)upstream $(tput sgr0) git@github.com:example-fork/fzf-zsh-completions (fetch)"
+    }
+
+    prefix=
+    _fzf_complete_git 'git pull --rebase '
+}
+
+@test 'Testing completion: git pull --shallow-exclude=**' {
+    _fzf_complete() {
+        assert $# equals 2
+        assert $1 same_as '--ansi --tiebreak=index '
+        assert $2 same_as 'git pull '
+
+        run cat
+        assert ${#lines} equals 5
+        assert ${lines[1]} same_as "$(tput setaf 3)--shallow-exclude=another-branch $(tput sgr0)  2nd commit"
+        assert ${lines[2]} same_as "$(tput setaf 3)--shallow-exclude=master         $(tput sgr0) 1st commit"
+        assert ${lines[3]} same_as "$(tput setaf 3)--shallow-exclude=v1             $(tput sgr0) 1st commit"
+        assert ${lines[4]} same_as "$(tput setaf 3)--shallow-exclude=v2             $(tput sgr0)  2nd commit"
+        assert ${lines[5]} same_as "$(tput setaf 3)--shallow-exclude=3e209a3        $(tput sgr0) 1st commit"
+    }
+
+    prefix=--shallow-exclude=
+    _fzf_complete_git 'git pull '
+}
+
+@test 'Testing completion: git pull --shallow-exclude **' {
+    _fzf_complete() {
+        assert $# equals 2
+        assert $1 same_as '--ansi --tiebreak=index '
+        assert $2 same_as 'git pull --shallow-exclude '
+
+        run cat
+        assert ${#lines} equals 5
+        assert ${lines[1]} same_as "$(tput setaf 3)another-branch $(tput sgr0)  2nd commit"
+        assert ${lines[2]} same_as "$(tput setaf 3)master         $(tput sgr0) 1st commit"
+        assert ${lines[3]} same_as "$(tput setaf 3)v1             $(tput sgr0) 1st commit"
+        assert ${lines[4]} same_as "$(tput setaf 3)v2             $(tput sgr0)  2nd commit"
+        assert ${lines[5]} same_as "$(tput setaf 3)3e209a3        $(tput sgr0) 1st commit"
+    }
+
+    prefix=
+    _fzf_complete_git 'git pull --shallow-exclude '
+}
+
+@test 'Testing completion: git pull --negotiation-tip=**' {
+    _fzf_complete() {
+        assert $# equals 2
+        assert $1 same_as '--ansi --tiebreak=index '
+        assert $2 same_as 'git pull '
+
+        run cat
+        assert ${#lines} equals 2
+        assert ${lines[1]} same_as --negotiation-tip=commit
+        assert ${lines[2]} same_as --negotiation-tip=glob
+    }
+
+    prefix=--negotiation-tip=
+    _fzf_complete_git 'git pull '
+}
+
+@test 'Testing completion: git pull --negotiation-tip **' {
+    _fzf_complete() {
+        assert $# equals 2
+        assert $1 same_as '--ansi --tiebreak=index '
+        assert $2 same_as 'git pull --negotiation-tip '
+
+        run cat
+        assert ${#lines} equals 2
+        assert ${lines[1]} same_as commit
+        assert ${lines[2]} same_as glob
+    }
+
+    prefix=
+    _fzf_complete_git 'git pull --negotiation-tip '
+}
+
+@test 'Testing completion: git pull --gpg-sign=**' {
+    skip $'This completion isn\'t implemented yet'
```

Comment:
Remove those skipped tests whose implementation has not yet been provided.

Reply:
Should I delete this also?
```zsh
# src/git.zsh
          --gpg-sign|-S)
                ;;
```

Reply:
Let's leave them as they are!